src/main/java/minimalisp/Lisp.java

```diff
@@ -341,6 +341,15 @@ public int compare(T o1, T o2) {
   }
 
   /**
+   * Returns a new Map with the contents of the second provided map merged over the provided map  
+   */
+  public static <K, V> Map<K, V> merge(Map<K, V> a, Map<K, V> b) {
+    Map<K, V> merged = new LinkedHashMap<K, V>(a);
```

Comment:
@martinpllu 

I think being able to get a different type of Map is a good idea.

As you propose, we could have a pass in the target Map as a third argument, but I think that there are simpler alternatives. 
**Instead, If you want to specify a type, why just have a function?** i.e.

```
// TreeMap keys are sorted
assertEquals(
  list("A", "B", "C", "D"),
  list(
    treeMap(
      merge(map("C", 10, "D", 11), map("A", 1, "B", 2, "C", 3)))
        .keySet()));  
```
and instead of creating an overloaded every function that creates a Map from a Map, we would just need:
```
  public static <K, V> Map<K, V> treeMap(Map<K, V> map) {
    return new TreeMap<K, V>(map);
  }
```
This is no extra cognitive work (actually less so) than passing the desired implementation of the `Map` in as a parameter, and beyond TreeMap (possibly ConcurrentHashMap) I don't see too much call for other functions (e.g. no point having a linkedHashMap function as insertion order has already been lost). 

The abstraction leaks well anyway - want a HashMap - just create a new one with the map as a parameter. 

But that still leaves the door open to introduce type construction based on parameters. Concretely, if you pass in TreeMaps you get a TreeMap back out e.g:
```
    // since the parameters are TreeMaps, 
    // the merged Map be should still have TreeMap sorted keys
    assertEquals(
        list("A", "B", "C", "D"),
        list(
            merge(treeMap("C", 10, "D", 11), treeMap("A", 1, "B", 2, "C", 3))
            .keySet()));  
```

In my eyes, that would be an expression of convention over configuration.

In which case the `merge` function (along with all other functions that construct a `Map` from another `Map`) would then become something like:
```
  public static <K, V> Map<K, V> merge(Map<K, V> a, Map<K, V> b) {
    Map<K, V> merged = newMapWithSameImplementationAs(a);
    merged.putAll(a);
    merged.putAll(b);
    return merged;
  }
```

What do you think? Does that seem sensible? 

----
FWIW - 

I think that in most cases one should stick with `LinkedHashMap`. Folks often jump to `TreeMap` because `HashMap` has no sorting at all. 

By preserving insertion order (which is often sorted anyway) `LinkedHashMap` makes it easy to predictably shift between Lists and Maps back again, plus it is almost as fast as a `HashMap`.